src/System.Private.CoreLib/shared/System/SpanHelpers.T.cs

```diff
@@ -51,54 +50,69 @@ public static int IndexOf<T>(ref T searchSpace, int searchSpaceLength, ref T val
         }
 
         // Adapted from IndexOf(...)
-        public static bool Contains<T>(ref T searchSpace, T value, int length)
+        public unsafe static bool Contains<T>(ref T searchSpace, T value, int length)
                where T : IEquatable<T>
         {
             Debug.Assert(length >= 0);
 
             IntPtr index = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
-            while (length >= 8)
-            {
-                length -= 8;
 
-                if (value.Equals(Unsafe.Add(ref searchSpace, index + 0)) ||
-                    value.Equals(Unsafe.Add(ref searchSpace, index + 1)) ||
-                    value.Equals(Unsafe.Add(ref searchSpace, index + 2)) ||
-                    value.Equals(Unsafe.Add(ref searchSpace, index + 3)) ||
-                    value.Equals(Unsafe.Add(ref searchSpace, index + 4)) ||
-                    value.Equals(Unsafe.Add(ref searchSpace, index + 5)) ||
-                    value.Equals(Unsafe.Add(ref searchSpace, index + 6)) ||
-                    value.Equals(Unsafe.Add(ref searchSpace, index + 7)))
+            if (default(T) != null || (object)value != null)
```

Comment:
Do you plan to add the null tests in corefx? Let's be sure that the NullReference issue has been resolved for all the MemoryExtension APIs.

Reply:
Might be a while; which is why I said "For dotnet/corefx#33710" rather than "Fixes" (so it didn't auto-close the issue) 😉 

As it would still need tests before it was complete.

Reply:
> Might be a while

That's perfectly fine. I can help with the test coverage (next week), if needed.

Reply:
It looks like tests for Spans of reference types (or T?) might be a bit light in general, are some arrays of string, but most T tests look to be `TInt` which is also a struct?

Reply:
Just `string` is probably fine; just meant there are less tests to get the coverage for the full unrolled exit points; whereas byte and char have a broader range (though being vectorized and depending on alignment, their number of exit points depending on input is even larger; which is likely why there are many more tests)

However, I only glanced so I could be wrong.